src/id.rs

```diff
@@ -86,12 +101,17 @@ impl parsec::SecretId for FullId {
         self.sign(data)
     }
 
-    fn encrypt<M: AsRef<[u8]>>(&self, _to: &Self::PublicId, _msg: M) -> Option<Vec<u8>> {
-        unimplemented!()
+    fn encrypt<M: AsRef<[u8]>>(&self, to: &Self::PublicId, plaintext: M) -> Option<Vec<u8>> {
+        serialise(&to.public_encryption_key().encrypt(plaintext)).ok()
     }
 
-    fn decrypt(&self, _from: &Self::PublicId, _ct: &[u8]) -> Option<Vec<u8>> {
-        unimplemented!()
+    fn decrypt(&self, _from: &Self::PublicId, ciphertext: &[u8]) -> Option<Vec<u8>> {
+        let ciphertext: encryption::Ciphertext = deserialise(ciphertext).ok()?;
+        if !ciphertext.verify() {
```

Comment:
Why are we doing that?
I expected it would be done by the decrypt function as otherwise it would not be safe.

```
    /// Returns the decrypted text, or `None`, if the ciphertext isn't valid.
    pub fn decrypt(&self, ct: &Ciphertext) -> Option<Vec<u8>> {
        if !ct.verify() {
            return None;
        }
        let Ciphertext(ref u, ref v, _) = *ct;
        let g = u.into_affine().mul(*self.0);
        Some(xor_with_hash(g, v))
    }
```

Reply:
Fixed in later commit

Reply:
This is outdated change.